Python/Include/Range.hpp

```diff
@@ -0,0 +1,68 @@
+#ifndef ROVER_PYTHON_RANGE_HPP
+#define ROVER_PYTHON_RANGE_HPP
+#include <string_view>
+#include <pybind11/pybind11.h>
+#include "Arithmetics.hpp"
+#include "Rover/Box.hpp"
+#include "Rover/Range.hpp"
+
+namespace Rover {
+
+  //! Exports an instantiation of the Range class.
+  /*!
+    \param module The module to export the class to.
+    \param type_name The name of the type the Range generates.
+  */
+  template<typename B, typename E, typename G>
+  void export_range(pybind11::module& module, std::string_view type_name);
+
+  //! Exports a generic Range for Python objects.
+  void export_range(pybind11::module& module);
+
+  template<typename Begin, typename End, typename Granularity>
+  void export_range(pybind11::module& module, std::string_view type_name) {
+    auto name = std::string("Range").append(type_name);
+    if constexpr(std::is_same_v<Granularity, void>) {
+      pybind11::class_<Range<Begin, End>>(module, name.c_str())
+        .def(pybind11::init(
+          [] (pybind11::object begin, pybind11::object end) {
+            return std::make_unique<Range<Begin, End>>(begin.cast<Begin>(),
+              end.cast<End>());
+          }))
+        .def("generate", &Range<Begin, End, Granularity>::generate);
+    } else {
+      pybind11::class_<Range<Begin, End, Granularity>>(module, name.c_str())
+        .def(pybind11::init(
+          [] (pybind11::object begin, pybind11::object end,
+              pybind11::object granularity) {
+            return std::make_unique<Range<Begin, End, Granularity>>(
+              begin.cast<Begin>(), end.cast<End>(),
+              granularity.cast<Granularity>());
+          }))
+        .def("generate", &Range<Begin, End, Granularity>::generate);
+    }
+    pybind11::implicitly_convertible<Range<Begin, End, Granularity>,
+      Box<Range<Begin, End, Granularity>::Type>>();
+    pybind11::implicitly_convertible<Box<Range<Begin, End,
+      Granularity>::Type>, Range<Begin, End, Granularity>>();
```

Comment:
There is no way to implicitly convert from a ```Box``` to a ```Range```, although converting from a ```Range``` to a ```Box``` is absolutely supported.